# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsRange.h
# Commit: c1e64e8e4665
# Full Hash: c1e64e8e466506d1d9a36d747df25527bd545b10
# Author: Martin Robinson <mrobinson@igalia.com>
# Date: 2022-12-01 16:18:29
# Regressor Bug: 1791759
# File Overlap Count: 1
# Description:
#   Bug 1791759 - Add support for `content-visibility: auto` r=emilio
#   
#   This change adds support for `content-visibilty: auto` as well as
#   showing and hiding content based on the relevancy of the content as
#   defined in the specification. Changes to relevancy are handled by
# ==============================================================================

diff -r 6d4d56f57b7b -r c1e64e8e4665 dom/base/nsRange.h
--- a/dom/base/nsRange.h	Thu Dec 01 01:23:08 2022 +0200
+++ b/dom/base/nsRange.h	Wed Nov 30 18:00:27 2022 +0000
@@ -353,6 +353,18 @@
    */
   MOZ_CAN_RUN_SCRIPT void NotifySelectionListenersAfterRangeSet();
 
+  /**
+   * For a range for which IsInSelection() is true, return the closest common
+   * inclusive ancestor
+   * (https://dom.spec.whatwg.org/#concept-tree-inclusive-ancestor)
+   * for the range, which we had to compute when the common ancestor changed or
+   * IsInSelection became true, so we could register with it. That is, it's a
+   * faster version of GetClosestCommonInclusiveAncestor that only works for
+   * ranges in a Selection. The method will assert and the behavior is undefined
+   * if called on a range where IsInSelection() is false.
+   */
+  nsINode* GetRegisteredClosestCommonInclusiveAncestor();
+
  protected:
   /**
    * https://dom.spec.whatwg.org/#concept-tree-inclusive-ancestor
@@ -386,18 +398,6 @@
       const mozilla::RangeBoundaryBase<EPT, ERT>& aEndBoundary,
       nsINode* aRootNode, bool aNotInsertedYet = false);
 
-  /**
-   * For a range for which IsInSelection() is true, return the closest common
-   * inclusive ancestor
-   * (https://dom.spec.whatwg.org/#concept-tree-inclusive-ancestor)
-   * for the range, which we had to compute when the common ancestor changed or
-   * IsInSelection became true, so we could register with it. That is, it's a
-   * faster version of GetClosestCommonInclusiveAncestor that only works for
-   * ranges in a Selection. The method will assert and the behavior is undefined
-   * if called on a range where IsInSelection() is false.
-   */
-  nsINode* GetRegisteredClosestCommonInclusiveAncestor();
-
   // Assume that this is guaranteed that this is held by the caller when
   // this is used.  (Note that we cannot use AutoRestore for mCalledByJS
   // due to a bit field.)